src/main/scala/centaur/test/metadata/WorkflowMetadata.scala

```diff
@@ -11,14 +13,15 @@ import configs.Result
 import scala.util.{Failure, Success, Try}
 
 case class WorkflowMetadata(value: Map[String, JsValue]) extends AnyVal {
-  def diff(other: WorkflowMetadata): Iterable[String] = {
+
+  def diff(other: WorkflowMetadata, workflowID: UUID): Iterable[String] = {
     val missingErrors = value.keySet.diff(other.value.keySet) map { k => s"Missing key: $k" }
-    val mismatchErrors = value.keySet.intersect(other.value.keySet) flatMap { k => diffValues(k, value(k), other.value(k)) }
+    val mismatchErrors = value.keySet.intersect(other.value.keySet) flatMap { k => diffValues(k, value(k), other.value(k), workflowID.toString) }
 
     mismatchErrors ++ missingErrors
   }
 
-  private def diffValues(key: String, expected: JsValue, other: JsValue): Option[String] = {
+  private def diffValues(key: String, expected: JsValue, other: JsValue, workflowID: String): Option[String] = {
```

Comment:
Yeah. It makes it more clear what's going on from the signature. It also makes the function more flexible as really all you need is the ID so anything which can supply an ID can use it.